### PR TITLE
Fix incosistent hashing

### LINKED ISSUE #647: Inconsistent hashing
The hashing is not consistent when empty files are present. For more information take a look at the description of https://github.com/valory-xyz/governatooorr/pull/39.

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR fixes the inconsistent hashing issues by
- Not replacing CRLF with LF
- Raising early if an empty file is found when hashing

## Fixes

Fixes #647

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/helpers/ipfs/base.py [hunk 1]
```diff
@@ -43,35 +41,13 @@
...
 
 
-def _dos2unix(file_content: bytes) -> bytes:
-    """
-    Replace occurrences of Windows line terminator CR/LF with only LF.
-
-    :param file_content: the content of the file.
-    :return: the same content but with the line terminator
-    """
-    return re.sub(b"\r\n", b"\n", file_content, flags=re.M)
```
Comment (angrybayblade) [inline aea/helpers/ipfs/base.py:53]: This is the logic which caused the inconsistent hash issues when locking and publishing packages, not sure why this was here in the first place but removing this logic fixes the issue and does not cause any new issue AFAIC so can be removed safely

File: aea/helpers/ipfs/base.py [hunk 2]
```diff
@@ -43,35 +41,13 @@
...
-    is_text = _is_text(file_path)
     with open(file_path, "rb") as file:
-        file_b = file.read()
-        if is_text:
-            file_b = _dos2unix(file_b)
-
-    return file_b
+        data = file.read()
+    if len(data) == 0:
+        raise ValueError(f"File cannot be empty: {file_path}")
```
Comment (solarw) [inline aea/helpers/ipfs/base.py:49]: @angrybayblade why can not be empty?  looks like empty file is a common case.

File: aea/helpers/ipfs/base.py [hunk 3]
```diff
@@ -43,35 +41,13 @@
...
-    try:
-        with open_file(file_path, "r") as f:
-            f.read()
-        return True
-    except UnicodeDecodeError:
-        return False
-
-
 def _read(file_path: str) -> bytes:
     """Read a file, replacing Windows line endings if it is a text file."""
```
Comment (solarw) [inline aea/helpers/ipfs/base.py:45]: @angrybayblade i dont see any fix for windows line endings anymore, file is opened in binary mode, so python will preserve all chars. 

probably doc string is needed to be updated?

**Review by solarw — COMMENTED** [review]
some points are unclear for me

File: aea/helpers/ipfs/base.py (hunk 2 above)
Comment (angrybayblade) [reply aea/helpers/ipfs/base.py:49]: The IPFS daemon treats empty files differently, if you push an empty file to the IPFS node it'll produce different hash compared to using the IPFSHashTool, now the reason behind this unclear I tried looking at the IPFS docs but could not find anything related. So for now we just raise an error when a user tries to hash an empty file so they don't run into these mismatching hashes issue when publishing the packages

File: aea/helpers/ipfs/base.py (hunk 3 above)
Comment (angrybayblade) [reply aea/helpers/ipfs/base.py]: Fixed [fb89dfe](https://github.com/valory-xyz/open-aea/pull/674/commits/fb89dfea890f1e1a2afa51e83843a001c7610625)

File: aea/helpers/ipfs/base.py (hunk 2 above)
Comment (solarw) [reply aea/helpers/ipfs/base.py:49]: @angrybayblade ok, let stay as is.
but i see a possible source of problems, if empty file will be needed for some reason

**Review by solarw — APPROVED** [review]
LGTM

**Review by solarw — APPROVED** [review]
lgtm

**Pull request merged**